lib/streamlit/type_util.py

```diff
@@ -264,14 +265,24 @@ def _is_probably_plotly_dict(obj: object) -> TypeGuard[dict[str, Any]]:
     return False
 
 
+def is_delta_generator(obj: object) -> TypeGuard[DeltaGenerator]:
+    return is_type(obj, "streamlit.delta_generator.DeltaGenerator")
```

Comment:
nit: the reason we check via the string is to avoid circular imports right? If so, might be worth a comment